pkg/operator/ceph/csi/csi.go

```diff
@@ -267,11 +267,17 @@ func (r *ReconcileCSI) setParams(ver *version.Info) error {
 		}
 	}
 
+	// EnableLeaderElection is true by default
+	CSIParam.EnableLeaderElection = true
+
 	CSIParam.ProvisionerReplicas = defaultProvisionerReplicas
 	nodes, err := r.context.Clientset.CoreV1().Nodes().List(r.opManagerContext, metav1.ListOptions{})
 	if err == nil {
 		if len(nodes.Items) == 1 {
 			CSIParam.ProvisionerReplicas = 1
+			// If there is only one node, we don't need to enable leader
+			// election for the provisioner pods.
+			CSIParam.EnableLeaderElection = false
```

Comment:
What is the side effect of running leader election when there is only a single provisioner running? Seems like it's just a no-op, right? Seems like there is no need to even disable it.

Reply:
Even with a single node, a terminating provisioner may still handle requests, and a newly started provisioner should not try to handle requests before the old provisioner is really finished. There needs to be some locking for it to be safe, the leader election is a nice option for that. I recommend to enable leader election in all situations.

Reply:
if the leader election is enabled when we have a single provisioner i need to wait for almost 2 minutes to take over the request which is not expected from user point of view. i dropped this commit for now.